src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -272,6 +272,30 @@ def get_published_sentry_apps_for_organization(
         )
         return [serialize_sentry_app(app) for app in published_apps]
 
+    def get_internal_integrations(
+        self, *, organization_id: int, integration_name: str
+    ) -> list[RpcSentryApp]:
+        """
+        Get all internal integrations for an organization matching a specific name.
+
+        Internal integrations are Sentry Apps that are created for use within a single
+        organization and are not available to be installed by users.
+
+        Args:
+            organization_id (int): The ID of the organization to search within
+            integration_name (str): The name of the internal integration to find
+
+        Returns:
+            list[RpcSentryApp]: A list of serialized internal Sentry Apps matching the criteria.
+                               Returns an empty list if no matches are found.
+        """
+        internal_integrations = SentryApp.objects.filter(
+            owner_id=organization_id,
+            status=SentryAppStatus.INTERNAL,
+            name=integration_name,
+        )
```

Comment:
Will each organization be getting a distinct sentryapp? Or will there be one sentry app for vercel and each organization has an installation?

Reply:
Each organization will have their own Vercel SentryApp. We do this since the access token tied to each organization will be stored in the SentryApp's metadata.